File: weyl-core/src/main/java/io/quartic/weyl/core/model/LayerSpec.java
package io.quartic.weyl.core.model;

import io.quartic.common.SweetStyle;
import io.quartic.weyl.core.live.LayerView;
import org.immutables.value.Value;

@SweetStyle
@Value.Immutable
public interface LayerSpec {
    LayerId id();
    LayerMetadata metadata();
    LayerView view();
    StaticSchema staticSchema();
    boolean indexable();
}

File: weyl-core/src/main/kotlin/io/quartic/weyl/core/model/LayerSpec.kt
package io.quartic.weyl.core.model

import io.quartic.weyl.core.live.LayerView

data class LayerSpec(
    val id: LayerId,
    val metadata: LayerMetadata,
    val view: LayerView,
    val staticSchema: StaticSchema,
    val indexable: Boolean
)
